Experiments/Experiments/CachedABTestVariationProvider.swift

```diff
@@ -0,0 +1,27 @@
+import AutomatticTracks
+
+/// Cache based implementation of `ABTestVariationProvider`
+///
+public struct CachedABTestVariationProvider: ABTestVariationProvider {
+
+    private let cache: VariationCache
+
+    public init(cache: VariationCache = VariationCache(userDefaults: .standard)) {
+        self.cache = cache
+    }
+
+    public func variation(for abTest: ABTest) -> Variation {
+        guard abTest.context == .loggedOut else {
+            return abTest.variation ?? .control
+        }
+
+        if let cachedVariation = cache.variation(for: abTest) {
+            return cachedVariation
+        } else if let variation = abTest.variation {
+            try? cache.assign(variation: variation, for: abTest)
+            return variation
+        } else {
+            return .control
+        }
```

Comment:
I think we should priority fresh values, i.e. getting the variation from `abTest` and only use `cache` as the fallback. That way we always get the latest values:
```suggestion
        if let variation = abTest.variation {
            try? cache.assign(variation: variation, for: abTest)
            return variation
        } else if let cachedVariation = cache.variation(for: abTest) {
            return cachedVariation
        } else {
            return .control
        }
```

Reply:
Done in https://github.com/woocommerce/woocommerce-ios/pull/8806/commits/c0708bfcf930776dac5b95cd6236829c69d95e22

Earlier, I prioritised cached value to display that we return the cached value if it is available. Because, for a logged-out experiment, it is perfectly valid to get the variation value only once and use it without refresh.